code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -127,24 +133,10 @@ class AEPMessage implements FullscreenMessage {
     @SuppressLint("ResourceType")
     @Override
     public void show() {
-        if (messagesMonitor != null && messagesMonitor.isDisplayed()) {
-            Log.debug(
-                    ServiceConstants.LOG_TAG,
-                    TAG,
-                    "Message couldn't be displayed, another message is displayed at this time.");
-            fullScreenMessageDelegate.onShowFailure();
-            return;
-        }
-
-        if (!fullScreenMessageDelegate.shouldShowMessage(this)) {
-            Log.debug(
-                    ServiceConstants.LOG_TAG,
-                    TAG,
-                    "Message couldn't be displayed, FullscreenMessageDelegate#shouldShowMessage"
-                            + " states the message should not be displayed.");
-            return;
-        }
+        show(true);
+    }
 
+    void show(final boolean withDelegateControl) {
```

Comment:
Do we have a use case where we might call `show(false)`?

Reply:
`show(true)` will be called when the Messaging extension creates an AEPMessage object then tries to show it. This call occurs within the Messaging extension. `show(false)` is called from within a customer's FullscreenMessageDelegate implementation. A customer can store the AEPMessage and decide to show it at a later time by calling AEPMessage.show().

Reply:
AEPMessage class is not public. If customers need to call show(false) then the method should be added to https://github.com/adobe/aepsdk-core-android/blob/dev-v2.0.0/code/core/src/main/java/com/adobe/marketing/mobile/services/ui/FullscreenMessage.java#L18